kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/datatest/names.kt

```diff
@@ -0,0 +1,44 @@
+package io.kotest.core.datatest
+
+import io.kotest.mpp.bestName
+import io.kotest.mpp.reflection
+import kotlin.reflect.KClass
+import kotlin.reflect.KType
+
+/**
+ * Each test name must be unique. We can use the toString if we determine the instance is stable.
+ *
+ * An instance is considered stable if it is a data class where each parameter is either a data class itself,
+ * or a primitive type.
+ *
+ * Note: If the user has overridden toString() and the returned value is not stable, tests may not appear.
+ */
+internal fun stableIdentifier(t: Any, counter: Int): String {
+   return if (isStable(t::class)) {
+      t.toString()
+   } else {
+      t::class.bestName() + " $counter"
+   }
+}
+
+internal fun isStable(type: KType) = when (val classifier = type.classifier) {
+   is KClass<*> -> isStable(classifier)
+   else -> false
+}
+
+internal fun isStable(type: KClass<*>): Boolean {
+   return when (type) {
+      String::class -> true
+      Int::class -> true
+      Long::class -> true
+      Double::class -> true
+      Float::class -> true
+      Byte::class -> true
+      Short::class -> true
+      Boolean::class -> true
```

Comment:
Can we simplify it somehow? 
As an option, we can have a set of classes that we consider as stable and check if `type` is in the set or call `javaPrimitiveType != null`  or `klass.java.isPrimitive`

Reply:
I didn't know about klass.java.isPrimitive so I will use that instead.

Reply:
Ah that's not available on multiplatform, since it's Java only.

Reply:
I moved the stable types to a set.